core/events/events_click.ts

```diff
@@ -60,9 +61,9 @@ export class Click extends UiBase {
    *
    * @returns JSON representation.
    */
-  override toJson(): AnyDuringMigration {
-    const json = super.toJson();
-    json['targetType'] = this.targetType;
+  override toJson(): ClickJson {
+    const json = super.toJson() as ClickJson;
+    json['targetType'] = this.targetType!;
```

Comment:
i don't understand why this one is safe. in the constructor it might be undefined since it's optional, so how do we know it's not undefined here? it seems like either we should set a default value or we should mark it optional in the json interface